src/Pact/Analyze/Types/Languages.hs

```diff
@@ -248,6 +256,48 @@ data Prop a
   | CoreProp     (Core Prop a)
   deriving (Show, Eq)
 
+instance UserShow a => UserShow (PropSpecific a) where
+  userShowsPrec _d = \case
+    Abort                   -> "abort"
+    Success                 -> "success"
+    Result                  -> "result"
+    Forall _ var ty x       -> parenList ["forall", parens (var <> ":" <> userShow ty), userShow x]
+    Exists _ var ty x       -> parenList ["exists", parens (var <> ":" <> userShow ty), userShow x]
+    TableWrite tab          -> parenList ["table-write", userShow tab]
+    TableRead  tab          -> parenList ["table-read", userShow tab]
+    ColumnWrite tab col     -> parenList ["column-write", userShow tab, userShow col]
+    ColumnRead tab col      -> parenList ["column-read", userShow tab, userShow col]
+    IntCellDelta tab col rk -> parenList ["int-cell-delta", userShow tab, userShow col, userShow rk]
+    DecCellDelta tab col rk -> parenList ["dec-cell-delta", userShow tab, userShow col, userShow rk]
+    IntColumnDelta tab col  -> parenList ["int-column-delta", userShow tab, userShow col]
+    DecColumnDelta tab col  -> parenList ["dec-column-delta", userShow tab, userShow col]
+    RowRead tab rk          -> parenList ["row-read", userShow tab, userShow rk]
+    RowReadCount tab rk     -> parenList ["row-read-count", userShow tab, userShow rk]
+    RowWrite tab rk         -> parenList ["row-write", userShow tab, userShow rk]
+    RowWriteCount tab rk    -> parenList ["row-write-count", userShow tab, userShow rk]
+
+instance UserShow a => UserShow (Prop a) where
+  userShowsPrec d = \case
+    PropSpecific p -> userShowsPrec d p
+    CoreProp     p -> userShowsPrec d p
+
+instance UserShow a => UserShow (Core Prop a) where
+  -- TODO: this is exactly duplicated with `UserShow (Core Term a)`
+  userShowsPrec d = \case
+    Lit a                    -> userShowsPrec d a
+    Sym s                    -> tShow s
+    StrConcat x y            -> parenList ["+", userShow x, userShow y]
```

Comment:
@bts I would like to convert all of these to, eg, `parenList [SAddition, userShow x, userShow y]`. Unfortunately this induces a circular import:

```
Module imports form a cycle:
         module `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
        imports `Pact.Analyze.Types' (/home/joel/code/github/kadena-i
o/pact/src/Pact/Analyze/Types.hs)
  which imports `Pact.Analyze.Types.Languages' (/home/joel/code/githu
b/kadena-io/pact/src/Pact/Analyze/Types/Languages.hs)
  which imports `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
```

It's not clear how I should avoid this without moving all instances into `Feature`. What do you think?